litex_boards/platforms/zedboard.py

```diff
@@ -0,0 +1,256 @@
+        "LA23_P": "E15",
+        "LA24_N": "A19",
+        "LA24_P": "A18",
+        "LA25_N": "C22",
+        "LA25_P": "D22",
+        "LA26_N": "E18",
+        "LA26_P": "F18",
+        "LA27_N": "D21",
+        "LA27_P": "E21",
+        "LA28_N": "A17",
+        "LA28_P": "A16",
+        "LA29_N": "C18",
+        "LA29_P": "C17",
+        "LA30_N": "B15",
+        "LA30_P": "C15",
+        "LA31_N": "B17",
+        "LA31_P": "B16",
+        "LA32_N": "A22",
+        "LA32_P": "A21",
+        "LA33_N": "B22",
+        "LA33_P": "B21",
+        "PRSNT_M2C_L": "AB14"
+    })
+]
+
+
+class Platform(XilinxPlatform):
+    default_clk_name = "clk100"
+    default_clk_period = 10.0
+
+    def __init__(self, toolchain="vivado"):
+        XilinxPlatform.__init__(self, "xc7z020clg484-1", _io, _connectors,
+                                toolchain=toolchain)
+        # self.toolchain.bitstream_commands = \
+        #     ["set_property BITSTREAM.CONFIG.SPI_BUSWIDTH 4 [current_design]"]
+        # self.toolchain.additional_commands = \
+        #     ["write_cfgmem -force -format bin -interface spix4 -size 16 "
+        #      "-loadbit \"up 0x0 {build_name}.bit\" -file {build_name}.bin"]
+        # self.toolchain.additional_commands = [
+        #     # Generate .bit.bin file for loading with linux fpga_manager
+        #     'exec bootgen -image {build_name}.bif -w -arch zynq -process_bitstream bin'
+        # ]
+
+    def add_oled(self, soc, SPI_N=0, SS_N=1, DC_GPIO=8, RST_GPIO=9):
+        '''
+        Wire-up the on-board OLED display to the Zynq PS
+
+        soc: a SocZynq object
+        SPI_N: which SPI peripheral (0 or 1)
+        SS_N: which slave select pin (0, 1, 2)
+        DC_GPIO: PS GPIO pin to connect to  Data / Command input of display
+        RST_GPIO: PS GPIO pin to connect to  Reset input of display
+
+        The configuration in ./ip/gen_ip.tcl must match these parameters!
+        '''
+        oled = self.request("zed_oled")
+        oled_cs = Signal()
+        soc.cpu.cpu_params[f"o_SPI{SPI_N}_SS{SS_N}_O"] = oled_cs
+        oled_clk = Signal()
+        oled_mosi = Signal()
+        soc.comb += [
+            # OLED power always on
+            oled.vbat_n.eq(0),
+            oled.vdd_n.eq(0),
+            # Fake the missing OLED chip select by gating MOSI and SCLK
+            If(oled_cs,
+                oled_clk.eq(0),
+                oled_mosi.eq(0)
+            ).Else(
+                oled_clk.eq(soc.cpu.cpu_params[f"o_SPI{SPI_N}_SCLK_O"]),
+                oled_mosi.eq(soc.cpu.cpu_params[f"o_SPI{SPI_N}_MOSI_O"]),
+            ),
+            # Share SPI0 SCLK and MOSI
+            oled.clk.eq(oled_clk),
+            oled.mosi.eq(oled_mosi),
+            # D/C = EMIO62
+            oled.dc.eq(soc.cpu.cpu_params["o_GPIO_O"][DC_GPIO]),
+            # RESET_N = EMIO63
+            oled.reset_n.eq(soc.cpu.cpu_params["o_GPIO_O"][RST_GPIO])
+        ]
+
+    def create_programmer(self, programmer="xc3sprog"):
```

Comment:
For simplicity, i only kept a default programmer for the others platforms. On 7-Series the default is generally OpenOCD which is fast/reliable and allow loading bitstream to SRAM but also flashing SPI Flashes. Can you choose and only keep one porgrammer?